File: DSAlgorithms/Leetcode/ts/src/stacks/946-validate_stack.ts
const validateStackSequences = (pushed: number[], popped: number[]): boolean =>{
    const a_stack: number[] = [];
    let j = 0;
    for(let i = 0; i < pushed.length; i++)
    {
        a_stack.push(pushed[i]);
        while(j < popped.length && a_stack.length > 0 && popped[j] === a_stack[a_stack.length - 1])
        {
            a_stack.pop();
            j++;
        }
    }
    return a_stack.length < 1;

};

File: DSAlgorithms/Leetcode/c++/stacks/946-validate_stack.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
       std::stack<int> a_stack;
       int j = 0;
       for(int i = 0; i < pushed.size(); i++) 
       {
           a_stack.push(pushed[i]);
           while(j < popped.size() && !a_stack.empty() && popped[j] == a_stack.top())
           {
               a_stack.pop();
               j++;
           }
       }
       return a_stack.empty();
    }
};